### PR TITLE
[Issue-195] Limit emails count per SMTP connection

### LINKED ISSUE #195: limit messages count per SMTP connection can send
#### Describe the feature

Support control the messages count per SMTP connection can send.  Close the SMTP connection when messages count limit reached.

#### Use cases

From [SendGrid SMTP Rate limits](https://docs.sendgrid.com/for-developers/sending-email/integrating-with-the-smtp-api#rate-limits)

> You may send up to 5k messages per SMTP connection.
> You may open up to 10k concurrent connections from a single server.

### PR DESCRIPTION
**Pull request by gaol**
Motivation: Fixes #195

Explain here the context, and why you're making that change, what is the problem you're trying to solve.

Conformance:

Your commits should be signed and you should have signed the Eclipse Contributor Agreement as explained in https://github.com/eclipse/vert.x/blob/master/CONTRIBUTING.md
Please also make sure you adhere to the code style guidelines: https://github.com/vert-x3/wiki/wiki/Vert.x-code-style-guidelines

### CODE REVIEW

**Comment by gaol** [comment]
Rejecting the mail submission without closing the connection will lead to the `bad` connection left in the connection pool if  the server does not close the connection(like wrong number configured). 

And rejecting mail submission may confuse user because I think what they really want is limiting the emails count per connection w/o failing sending emails.

**Comment by NeoCN** [comment]
@gaol hi, want to know when will this pr be merged and released ?

**Comment by gaol** [comment]
@vietj wdyt on this pr?

File: src/main/java/io/vertx/ext/mail/MailConfig.java [hunk 1]
```diff
@@ -173,6 +175,7 @@ public MailConfig(MailConfig other) {
     keepAliveTimeoutUnit = other.keepAliveTimeoutUnit;
     ntDomain = other.ntDomain;
     workstation = other.workstation;
+    countsPerConn = other.countsPerConn;
```
Comment (vietj) [inline src/main/java/io/vertx/ext/mail/MailConfig.java:178]: rename it `maxInflightMailsPerConnection`

Comment (gaol) [reply src/main/java/io/vertx/ext/mail/MailConfig.java]: Done :)

**Review by vietj — APPROVED** [review]

**Comment by vietj** [comment]
LGTM @gaol

**Pull request merged**